stdlib/record.ncl

```diff
@@ -107,5 +107,93 @@
         else
           r in
         %record_insert% field r content,
+    
+    map_values: forall a b. (a -> b) -> {_: a} -> {_: b}
+    | doc m%"
+      Maps a function over every field value of a record.
+
+      For example:
+      ```nickel
+        map_values (fun x => x + 1) { hi = 2 } =>
+          { hi = 3 }
+        map (fun x => x + 1) { hello = 1, world = 2 } =>
+          { hello = 2, world = 3 }
+      ```
+    "%
+    = fun f => map (fun _field value => f value),
+
+    as_array : forall a. {_: a} -> Array {field: Str, value: a}
+    | doc m%"
+      Converts a record into an array of key-value pairs.
+
+      For example:
+      ```nickel
+        as_array { hello = "world", foo = "bar" } =>
+          [ { field = "hello", value = "world" },
+            { field = "foo", value = "bar" }
+          ]
+      ```
+    "%
+    = fun record =>
+      record
+      |> fields
+      |> array.map (fun field_ => {field = field_, value = record."%{field}"}),
```

Comment:
Just a suggestion, but I always forget that we can use `'` as indentifiers. I find it prettier, but that's unimportant, and entirely subjective, so feel free to ignore :slightly_smiling_face: 

Beside, using `field'` instead of `field` makes the record non-recursive (technically, the thunk is non-revertible), which is probably better in general for performances.

```suggestion
      |> array.map (fun field' => {field = field', value = record."%{field'}"}),
```